src/pages/Inventory.page.js

```diff
@@ -23,6 +31,34 @@ export class InventoryPage extends BaseSwagLabPage {
         return this.inventoryItems.locator('div.pricebar > div').allTextContents();
     }
 
+    async productInfoByID(id) {
+        let product = new Object();
+        product.name;
+        product.description;
+        product.price;
+        
+        product.name = await this.inventoryItems.nth(id)
+        .locator('[data-test="inventory-item-name"]')
+        .innerText();
+    
+        product.description = await this.inventoryItems.nth(id)
+        .locator('[data-test="inventory-item-desc"]')
+        .innerText();
+    
+        product.price = await this.inventoryItems.nth(id)
+        .locator('div.pricebar > div')
+        .innerText();
+    
+        return product;
+    }
+
+    async addItemToCartByItemId(id) {
+        await this.inventoryItems.nth(id)
+        .filter({has: this.page.getByRole('button')})
```

Comment:
Why do we need this filter?

Reply:
I use this filter to get a button with further check for its property [data-test^="add-to-cart"] and click on it. Otherwise it can be other UI control with [data-test^="add-to-cart"] where click couldn't be possible or could lead to an unexpected result.